frontend/src/models/models.js

```diff
@@ -89,6 +90,41 @@ module.exports = app => app.component('models', {
     this.status = 'loaded';
   },
   methods: {
+    clickFilter(path) {
+       // Initialize searchFieldsMap if empty
+      if (!this.searchText) {
+        this.searchFieldsMap = {};
+      }
+
+      // Extract existing key-value pairs from searchText
+      const regex = /(\w+):\s*([^,}]*)/g;
```

Comment:
This won't quite work in all cases. For example, check out what happens when you try to click on a header with filter `{ _id: 'foo, bar' }`.